### DISCUSSION TITLE
Custom adapter raising "undefined method `memoize='"

### DISCUSSION THREAD

**Discussion by jamesgecko** [jnunemaker/flipper#555] [Q&A] [answered]
Hello! I'm trying to add feature flagging for a method that generates and caches some Javascript. To make the flag work, I need to rebuild the cache. I've created an adapter like so:

```ruby
require 'flipper'

class CacheBuster < SimpleDelegator
  include ::Flipper::Adapter

  attr_reader :name

  def initialize(adapter)
    @adapter = adapter
    @name = :cache_buster
  end

  def enable(feature, gate, thing)
    @adapter.enable(feature, gate, thing)
    rebuild_cache(thing)
  end

  def disable(feature, gate, thing)
    @adapter.disable(feature, gate, thing)
    rebuild_cache(thing)
  end

  private

  def rebuild_cache(thing)
    # Call the cache rebuild here.
  end
end
```
 And I'm setting it up like this:

```ruby
require 'flipper'
require 'flipper/adapters/redis'
require_relative '../../lib/cache_buster'

Flipper.configure do |config|
  config.default do
    CacheBuster.new(Flipper::Adapters::Redis.new($redis))
  end
end
```

But when I run my app, I get the following exception:
```
NoMethodError (undefined method `memoize=' for #<CacheBuster:0x00007ff4d7c9c808>):

/Users/intel/.rbenv/versions/2.6.6/lib/ruby/2.6.0/delegate.rb:87:in `method_missing'
/Users/intel/.rbenv/versions/2.6.6/lib/ruby/2.6.0/forwardable.rb:225:in `memoize='
flipper (0.22.0) lib/flipper/middleware/memoizer.rb:84:in `ensure in memoized_call'
flipper (0.22.0) lib/flipper/middleware/memoizer.rb:84:in `memoized_call'
flipper (0.22.0) lib/flipper/middleware/memoizer.rb:42:in `call'
rack (2.2.3) lib/rack/etag.rb:27:in `call'
rack (2.2.3) lib/rack/conditional_get.rb:27:in `call'
rack (2.2.3) lib/rack/head.rb:12:in `call'
(etc)
```

I'm confused. `memoize=` isn't mentioned in [the list of functions](https://github.com/jnunemaker/flipper/blob/master/docs/Adapters.md#roll-your-own) an adapter needs to implement. And shouldn't SimpleDelegator be passing it to the Redis adapter anyway?

**Comment by jamesgecko** [comment, marked as answer]
Nevermind, I overlooked the fact that you need to actually create an instance of Flipper in the configure block.

```ruby
Flipper.configure do |config|
  config.default do
    adapter = CacheBuster.new(Flipper::Adapters::Redis.new($redis))
    Flipper.new(adapter)
  end
end
```

Also need to define the methods passed through to the Redis class. 🤦‍♂️

**Reply by jnunemaker** [reply]
@jamesgecko yep, you got it. So sorry I didn't respond, though I see @bkeepers said your answer was correct. 

Somehow I never received an email notification for this. :( 

Lastly, if you don't mind, I would love to know what you are using flipper for as well and how it is working out for you (what went well, what was hard other than what you mentioned here).

If you aren't comfortable leaving that information here in a public comment, feel free to email me directly john@fewerandfaster.com. I'm just always curious. :)

Happy flipping!
